### PR TITLE
Reapply artifacts parser

### PR DESCRIPTION
**Pull request by pasinskim**

### CODE REVIEW

File: rootfs.go [hunk 1]
```diff
@@ -65,13 +66,10 @@ func doRootfs(device UInstaller, args runOptionsType, dt string) error {
 		log.Debugf("Feting update from file results: [%v], %d, %v", image, imageSize, err)
 	}
 
-	if image != nil {
-		defer image.Close()
-	}
-
-	if err != nil {
+	if image == nil || err != nil {
 		return errors.New("Error while updateing image from command line: " + err.Error())
```
Comment (bboozzoo) [inline rootfs.go]: You might want to fix _updateing_ typo.

File: rootfs.go [hunk 2]
```diff
@@ -65,13 +66,10 @@ func doRootfs(device UInstaller, args runOptionsType, dt string) error {
 		log.Debugf("Feting update from file results: [%v], %d, %v", image, imageSize, err)
 	}
 
-	if image != nil {
-		defer image.Close()
-	}
-
-	if err != nil {
+	if image == nil || err != nil {
```
Comment (bboozzoo) [inline rootfs.go]: Is it possible that both `err != nil` and `image != nil` ?

File: rootfs.go [hunk 3]
```diff
@@ -62,16 +62,17 @@ func doRootfs(device installer.UInstaller, args runOptionsType, dt string) error
...
 
 	if image == nil || err != nil {
 		return errors.Wrapf(err, "rootfs: error while updating image from command line")
 	}
 	defer image.Close()
 
+	fmt.Fprintf(os.Stdout, "Installing update from the artifact of size %d\n", imageSize)
 	p := &utils.ProgressWriter{
-		Out: os.Stderr,
+		Out: os.Stdout,
```
Comment (bboozzoo) [inline rootfs.go]: Why stdout? Shouldn't the progress be printed to stderr along with rest of the logs?

Comment (pasinskim) [inline rootfs.go]: Which logs are printed to stderr? I had the impression that those are printed to stdout. This was to unify the behavior.

Comment (bboozzoo) [inline rootfs.go]: I assumed that every log gets printed to stderr. Though, apparently, logrus defaults to stdout. In which case, you can leave the code as it is now.

**Comment by kacf** [comment]
Thanks, will run this in Jenkins.

**Comment by kacf** [comment]
https://ci.cfengine.com/job/yoctobuild-pr/94/

**Comment by kacf** [comment]
New one: https://ci.cfengine.com/job/yoctobuild-pr/97/

**Pull request merged**